react/README.md

```diff
@@ -127,6 +127,19 @@
       return WithFoo;
     }
     ```
+
+  - **Props Naming**: Avoid using DOM component prop names for different purposes.
```

Comment:
Although react/forbid-component-props is similar, it does not exactly enforce this. I'm not aware of any that do. It would be nice if there was one, though. ;)